src/components/chats/hooks/useLastMessageId.ts

```diff
@@ -0,0 +1,45 @@
+import { getCommentIdsByPostIdFromChainQuery } from '@/services/subsocial/commentIds'
+import { getPostMetadataQuery } from '@/services/subsocial/datahub/posts/query'
+import { getDatahubConfig } from '@/utils/env/client'
+
+export function useLastMessageId(chatId: string) {
+  const canUseDatahub = !!getDatahubConfig()
+
+  // need last message only
+  const { data: messageIds } = getCommentIdsByPostIdFromChainQuery.useQuery(
+    chatId,
+    {
+      subscribe: true,
+      enabled: !canUseDatahub,
+    }
+  )
+  const lastMessageId = messageIds?.[messageIds?.length - 1]
+
+  const { data } = getPostMetadataQuery.useQuery(chatId, {
+    enabled: canUseDatahub,
+  })
+  const lastIdFromDatahub = data?.lastCommentId
+
+  return canUseDatahub ? lastIdFromDatahub : lastMessageId
```

Comment:
Maybe use if(canUseDatahub), because hard to read, that the two different logic

Reply:
Same comment for other hooks

Reply:
Not rush, but add the comment into your tasks for future

Maybe create wrapers like on Polkaverse